### PR TITLE
Spanish optimization

### LINKED ISSUE #1416: Spanish keyboard missing buttons and functionality
### The Spanish keyboard is not very well thought out, and I have two suggestions.

The main concern here is that some of the most used signs in Spanish are really hidden. This signs are ¿ and ¡, which are used every time a question or an exclamation is made.

### 1. Here is where they are, in the second page of signs:
[](https://postimg.cc/McSrpxBK)
### 2. And they are not in the most common used section:
[](https://postimg.cc/hhZ2kVPr)

### Here is where I suggest they can be:
[](https://postimg.cc/mP2NkqnG)
[](https://postimg.cc/R6zTSF5g)

### PR DESCRIPTION
**Pull request by Fuseteam**
this is a recreation of #120 originally made by @J053Fabi0
it fixes https://github.com/ubports/ubuntu-touch/issues/1416

### CODE REVIEW

**Comment by Fuseteam** [comment]
bah why doesn't github recognize the original author in the commits :v

**Comment by Fuseteam** [comment]
Gottem :^3

**Comment by Fuseteam** [comment]
this fixes https://github.com/ubports/ubuntu-touch/issues/1416

**Comment by Bolly** [comment]
Hello, thank you for your hard work. I have some questions/suggestions.

I have to clarify that I have not installed the keyboard and I am only basing my opinion on this [picture](https://postimg.cc/DWyhGTFk).

**First question:**
I have seen two other keyboards (English and Catalan), and they are not as loaded as this one. Is there a reason from the design team for it to be that way? It seems more functional to me, and I find it odd that it is not implemented on the other two keyboards.

**Second question:**
Is there a design reason for having the numbers in the first row of the keyboard? Could they be removed? Can they be placed in several rows? 
In Spanish we have accents on vowels (**á, é í, ó** and **ú**), and four of the five vowels are in the first row.
When we write a number we probably need to write a number with more than one digit. So we switch keyboards **[?123]**, I don't think it's that important to have it so accessible.

**First suggestion:**
I find the following keyboard symbols missing: **¡ ! ¿ € **
And as they don't fit, I suggest removing some of these: **\* + = _**

**Second suggestion:**
As I hinted before, the first choice of vowel keys would have to be the same vowel with accent. (**á, é í, ó** and **ú**)

**Finally there is a repeated symbol:**
The **N** key and **,** key. They have the same symbol (**;**)

**Comment by Fuseteam** [comment]
hi @Bolly can you please verify by installing the keyboard if possible, the `¡ ! ¿ €` are for example available now
and the numbers shouldn't be the first choice right now on the top row as far as i can remember
i don't quite understand your first question however

**Comment by Bolly** [comment]
ok @Fuseteam . I should have installed it first. 😂 Forget my first question. 🙂

I find it confusing that sometimes the small print engraved on the key on the keyboard is the first to come out and sometimes it is not.

Examples:

1 -> First 1
2 -> First 2
3 -> First é
4 -> First 4
5 -> First 5
6 -> First 6
7 -> First ú
8 -> First í
9 -> First ó
0 -> First 0
@ -> First @
? -> First ¡
$ -> First £
....

cortao pr

The symbols "¡ ! €" are there. I was just hoping to have them closer at hand, but that's not important. Don't worry about it. 😊

**Comment by Fuseteam** [comment]
oh you're right the annotation should be corrected thanks, for the feedback, i'll try and fix it

**Comment by Bolly** [comment]
If you want the vowels (á, é í, ó and ú) to come out first. Here is a brainstorming session.

Propuesta 1
Propuesta 2

Or remove the recorded numbers. This is how it looks on the Ubuntu desktop.

Compact view.

Vista compacta

Phone view.

Vista telefono

And when you press the 12!@ key.




Or put the numbers together on only two keys of the keyboard. 

Example:

0...4 -> First 2 (0 1 **2** 3 4)
5...9 -> First 7 (5 6 **7** 8 9)

**Comment by Fuseteam** [comment]
oh my thanks for the pics, i'll see what i can do

File: plugins/es/qml/Keyboard_es_email.qml [hunk 1]
```diff
@@ -85,8 +85,8 @@ KeyPad {
             LanguageKey    { id: languageMenuButton;                     anchors.left: symShiftKey.right; height: parent.height; }
             CharKey        { id: atKey;    label: "@"; shifted: "@";     anchors.left: languageMenuButton.right; height: parent.height; }
             SpaceKey       { id: spaceKey;                               anchors.left: atKey.right; anchors.right: urlKey.left; noMagnifier: true; height: parent.height; }
-            UrlKey         { id: urlKey; label: ".es"; extended: [".com", ".mx", ".co", ".ar", ".pe", ".ve", ".cl"]; anchors.right: dotKey.left; height: parent.height; }
-            CharKey        { id: dotKey;      label: "."; shifted: "."; extended: ["?", "-", "_", "!", "+", "%","#","/"]; extendedShifted: ["?", "-", "_", "!", "+", "%","#","/"]; anchors.right: enterKey.left; height: parent.height; }
+            UrlKey         { id: urlKey; label: ".com"; extended: [".com", ".mx", ".es", ".org", ".net"]; anchors.right: dotKey.left; height: parent.height; }
```
Comment (Bolly) [inline plugins/es/qml/Keyboard_es_email.qml:88]: Why have the following domains been removed?
.co -> Colombia
.ar -> Argentina
.pe -> Peru
.ve -> Venezuela
.cl -> Chile

And these have been left?
.es -> España
.mx -> Mexico

Comment (Fuseteam) [reply plugins/es/qml/Keyboard_es_email.qml:88]: no idea why the original author removed those, i'll add them back, seems relevant

**Comment by peat-psuwit** [comment]
Hello,

Thank you for contributing to UBports. As part of project renaming and the effort to port Ubuntu Touch stack to Ubuntu 20.04, we're incrementally migrating repositories to GitLab.

Your PR is now migrated to:
https://gitlab.com/ubports/core/lomiri-keyboard/-/merge_requests/170
However, you might not be able to push new code into it, as the branch resides in our repository. In case you need to update it, please open a new MR from your (new) fork. I don't have a better solution at the moment.

Sorry for your inconvenience.

**Pull request closed without merging**